tensorflow/lite/micro/kernels/dynamic_update_slice_test.cc

```diff
@@ -0,0 +1,145 @@
+                             outputs_array, nullptr);
+
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, runner.InitAndPrepare());
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, runner.Invoke());
+}
+
+void TestDynamicUpdateSliceFloat(int* input_dims_data[kNumInputs],
+                                 const float* input_data_0,
+                                 const float* input_data_1,
+                                 const int32_t* input_data_2,
+                                 const float* golden_data, int* expected_dims,
+                                 float* output_data) {
+  TfLiteIntArray* input_dims_0 = IntArrayFromInts(input_dims_data[0]);
+  TfLiteIntArray* input_dims_1 = IntArrayFromInts(input_dims_data[1]);
+  TfLiteIntArray* input_dims_2 = IntArrayFromInts(input_dims_data[2]);
+  TfLiteIntArray* output_dims = IntArrayFromInts(expected_dims);
+  const int output_count = ElementCount(*output_dims);
+
+  TfLiteTensor tensors[] = {
+      CreateTensor(input_data_0, input_dims_0),
+      CreateTensor(input_data_1, input_dims_1),
+      CreateTensor(input_data_2, input_dims_2),
+      CreateTensor(output_data, output_dims),
+  };
+  constexpr int tensors_count = std::extent<decltype(tensors)>::value;
+  ExecuteDynamicUpdateSliceTest(tensors, tensors_count);
+
+  // check output data against expected
+  for (int i = 0; i < output_count; i++) {
+    printf("output_data[%d] = %f\n", i, output_data[i]);
+    TF_LITE_MICRO_EXPECT_NEAR(golden_data[i], output_data[i], 0.0);
+  }
+
+  // check output dimensions (relocated) against original dimensions
+  TF_LITE_MICRO_EXPECT_EQ(output_dims->size,
+                          tensors[kOutputTensorIndex].dims->size);
+  for (int i = 0; i < output_dims->size; i++) {
+    TF_LITE_MICRO_EXPECT_EQ(output_dims->data[i],
+                            tensors[kOutputTensorIndex].dims->data[i]);
+  }
+}
+
+// TODO(rameshkunasi): Add quantized test for dynamic update slice.
+
+}  // namespace
+}  // namespace testing
+}  // namespace tflite
+
+TF_LITE_MICRO_TESTS_BEGIN
+
+TF_LITE_MICRO_TEST(DynamicUpdateSliceOpTestSimpleFloat) {
+  int32_t kInputDims_0[] = {2, 3, 3};
+  int32_t kInputDims_1[] = {2, 2, 1};
+  int32_t kInputDims_2[] = {1, 2};
+  int32_t* kInputDims[tflite::testing::kNumInputs] = {
+      kInputDims_0, kInputDims_1, kInputDims_2};
+  int kOutputDims[] = {2, 3, 3};
+
+  constexpr float kInput_0[] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+  constexpr float kInput_1[] = {-1, -2};
+  constexpr int32_t kInput_2[] = {1, 1};
+  constexpr float kExpect[] = {1, 2, 3, 4, -1, 6, 7, -2, 9};
+  constexpr int kOutputCount = std::extent<decltype(kExpect)>::value;
+  float output_data[kOutputCount];
+
+  tflite::testing::TestDynamicUpdateSliceFloat(kInputDims, kInput_0, kInput_1,
+                                               kInput_2, kExpect, kOutputDims,
+                                               output_data);
+}
+
+TF_LITE_MICRO_TEST(DynamicUpdateSliceOpTestBoundaryTest) {
```

Comment:
What's the purpose of this test? How is this different from  `DynamicUpdateSliceOpTestSimpleFloat`?